File: build_info/lib/build_info.dart
// Copyright 2024 Craftsoft LLC. All rights reserved.
// Use of this source code is governed by a MIT-style license that can be
// found in the LICENSE file.

import 'package:build_info_platform_interface/build_info_data.dart';
import 'package:build_info_platform_interface/build_info_platform_interface.dart';

export 'package:build_info_platform_interface/build_info_data.dart'
    show BuildInfoData;

/// BuildInfo class
class BuildInfo {
  /// Get the [BuildInfoPlatform] instance.
  static BuildInfoPlatform get platform => BuildInfoPlatform.instance;

  /// Get [BuildInfoData] from the platform.
  static Future<BuildInfoData?> fromPlatform() {
    return platform.fromPlatform();
  }
}

File: build_info_android/android/src/main/kotlin/dev/craftsoft/buildinfo/BuildInfo.kt
// Copyright 2024 Craftsoft LLC. All rights reserved.
// Use of this source code is governed by a MIT-style license that can be
// found in the LICENSE file.

package dev.craftsoft.buildinfo

import android.app.Activity
import android.content.pm.PackageManager

class BuildInfo: BuildInfoHostApi {
    private var activity: Activity? = null

    fun setActivity(activity: Activity?) {
        this.activity = activity
    }

    override fun fromPlatform(): BuildInfoDataPigeon {
        val buildDateLong = BuildConfig.BUILDINFO_TIMESTAMP
        var installDateLong: Long? = null

        if (activity != null) {
            val packageName = activity!!.packageName

            val pm: PackageManager? = activity?.packageManager

            val pi = pm?.getPackageInfo(packageName, PackageManager.GET_ACTIVITIES)
            installDateLong = pi?.firstInstallTime
        }

        return BuildInfoDataPigeon(buildDateLong, installDateLong)
    }
}
